### PR TITLE
feat(generator-pie-component): DSW-1218 use createStory in component generator

### PR DESCRIPTION
**Pull request by xander-marjoram**
## Describe your changes (can list changeset entries if preferable)

- [Added] - use createStory in component generator
- [Fixed] - whitespace in generated story file

## Author Checklist (complete before requesting a review)
- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests
- [ ] If it is a `PIE Docs` change, I have reviewed the Docs site preview
- [ ] If it is a component change, I have reviewed the Storybook preview
- [ ] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### CODE REVIEW

File: packages/tools/generator-pie-component/src/app/templates/pie-placeholder.__stories__.ts [hunk 1]
```diff
@@ -31,11 +31,8 @@ export default <%= componentNameCamelCase %>StoryMeta;
 
 // TODO: remove the eslint-disable rule when props are added
 // eslint-disable-next-line no-empty-pattern
-const Template = ({ <% if (needsRTL) { %>dir<% } %> }: <%= componentName %>Props): TemplateResult => html`
+const Template = ({<% if (needsRTL) {%> dir <% }%>}: <%= componentName %>Props) => html`
```
Comment (xander-marjoram) [inline packages/tools/generator-pie-component/src/app/templates/pie-placeholder.__stories__.ts:34]: Previously the generated text was `{  }` or `{ dir }`, now it's `{}` or `{ dir }`.

File: packages/tools/generator-pie-component/src/app/templates/pie-placeholder.__stories__.ts [hunk 2]
```diff
@@ -15,9 +15,9 @@ const defaultArgs: <%= componentName %>Props = {};
 const <%= componentNameCamelCase %>StoryMeta: <%= componentName %>StoryMeta = {
     title: '<%= displayName %>',
     component: 'pie-<%= fileName %>',
-    argTypes: {
-    <% if (needsRTL) { %>...i18nArgTypes,<% } %>
-    },
+    argTypes: {<% if (needsRTL) { %>
```
Comment (xander-marjoram) [inline packages/tools/generator-pie-component/src/app/templates/pie-placeholder.__stories__.ts:18]: Previous the generated text had unnecessary line breaks when `needsRTL` was false, now it's `{}`.

File: packages/tools/generator-pie-component/src/app/templates/pie-placeholder.__stories__.ts [hunk 3]
```diff
@@ -1,9 +1,9 @@
-import { html, TemplateResult<% if (needsRTL) { %>, nothing <% } %>} from 'lit';
-import { type StoryObj as Story } from '@storybook/web-components';
+import { html<% if (needsRTL) { %>, nothing<% } %> } from 'lit';
 import { Pie<%= componentName %>, <%= componentName %>Props } from '@justeattakeaway/pie-<%= fileName %>';
 import { type StoryMeta } from '../types';
-<% if (needsRTL) { %>import { i18nArgTypes } from '../args/commonArgsTypes';<% } %>
-
+import { createStory } from '../utilities';
+<% if (needsRTL) { %>import { i18nArgTypes } from '../args/commonArgsTypes';
```
Comment (xander-marjoram) [inline packages/tools/generator-pie-component/src/app/templates/pie-placeholder.__stories__.ts:5]: When `needsRTL` was false, there was an extra line break

File: packages/tools/generator-pie-component/src/app/templates/pie-placeholder.__stories__.ts [hunk 4]
```diff
@@ -1,9 +1,9 @@
-import { html, TemplateResult<% if (needsRTL) { %>, nothing <% } %>} from 'lit';
-import { type StoryObj as Story } from '@storybook/web-components';
+import { html<% if (needsRTL) { %>, nothing<% } %> } from 'lit';
```
Comment (xander-marjoram) [inline packages/tools/generator-pie-component/src/app/templates/pie-placeholder.__stories__.ts:1]: Imports which are no longer required have been removed

**Review by raoufswe — APPROVED** [review]

**Review by siggerzz — APPROVED** [review]

**Pull request merged**